### PR TITLE
python312Packages.sparsezoo: init at 1.8.1

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

Add [python312Packages.sparsezoo](https://github.com/neuralmagic/sparsezoo), a neural network model repository.

- **python311Packages.py-machineid: init at 0.6.0**
- **python311Packages.sparsezoo: init at 1.8.1**

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 337214` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python311Packages.py-machineid</li>
    <li>python311Packages.py-machineid.dist</li>
    <li>python311Packages.sparsezoo</li>
    <li>python311Packages.sparsezoo.dist</li>
    <li>python312Packages.py-machineid</li>
    <li>python312Packages.py-machineid.dist</li>
    <li>python312Packages.sparsezoo</li>
    <li>python312Packages.sparsezoo.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 337214` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python311Packages.py-machineid</li>
    <li>python311Packages.py-machineid.dist</li>
    <li>python311Packages.sparsezoo</li>
    <li>python311Packages.sparsezoo.dist</li>
    <li>python312Packages.py-machineid</li>
    <li>python312Packages.py-machineid.dist</li>
    <li>python312Packages.sparsezoo</li>
    <li>python312Packages.sparsezoo.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 337214` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python311Packages.py-machineid</li>
    <li>python311Packages.py-machineid.dist</li>
    <li>python311Packages.sparsezoo</li>
    <li>python311Packages.sparsezoo.dist</li>
    <li>python312Packages.py-machineid</li>
    <li>python312Packages.py-machineid.dist</li>
    <li>python312Packages.sparsezoo</li>
    <li>python312Packages.sparsezoo.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 337214` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python311Packages.py-machineid</li>
    <li>python311Packages.py-machineid.dist</li>
    <li>python311Packages.sparsezoo</li>
    <li>python311Packages.sparsezoo.dist</li>
    <li>python312Packages.py-machineid</li>
    <li>python312Packages.py-machineid.dist</li>
    <li>python312Packages.sparsezoo</li>
    <li>python312Packages.sparsezoo.dist</li>
  </ul>
</details>

**Comment by natsukium** [comment]
result of `nixpkgs-lint --include-unfinished-lints pkgs/development/python-modules/{py-machineid,sparsezoo}` [1](https://github.com/natsukium/nixpkgs-lint/tree/620731fc7ec66b0c67e6b8ea039545f71ea78409)
```
Advice: redundant package in nativeBuildInputs
    ╭─[pkgs/development/python-modules/sparsezoo/default.nix:44:25]
    │
 44 │   nativeBuildInputs = [ pythonRelaxDepsHook ];
    │                       ───────────┬───────────  
    │                                  ╰───────────── remove this from nativeBuildInputs
    │                                  │             
    │                                  ╰───────────── part of this list
────╯

```

File: pkgs/development/python-modules/py-machineid/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    setuptools
+  ];
+
+  pythonImportsCheck = [ "machineid" ];
+
+  # Tests are not present in Pypi archive
+  doCheck = false;
+
+  meta = {
+    description = "Get the unique machine ID of any host (without admin privileges";
```
Comment (natsukium) [inline pkgs/development/python-modules/py-machineid/default.nix:28]:
```suggestion
    description = "Get the unique machine ID of any host (without admin privileges)";
```

File: pkgs/development/python-modules/py-machineid/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    homepage = "https://pypi.org/project/py-machineid/";
```
Comment (natsukium) [inline pkgs/development/python-modules/py-machineid/default.nix:29]: How about citing the GitHub repository?
https://github.com/keygen-sh/py-machineid

Comment (GaetanLepage) [reply pkgs/development/python-modules/py-machineid/default.nix:29]: Sure !

**Comment by GaetanLepage** [comment]
> result of `nixpkgs-lint --include-unfinished-lints pkgs/development/python-modules/{py-machineid,sparsezoo}` [1](https://github.com/natsukium/nixpkgs-lint/tree/620731fc7ec66b0c67e6b8ea039545f71ea78409)
> 
> ```
> Advice: redundant package in nativeBuildInputs
>     ╭─[pkgs/development/python-modules/sparsezoo/default.nix:44:25]
>     │
>  44 │   nativeBuildInputs = [ pythonRelaxDepsHook ];
>     │                       ───────────┬───────────  
>     │                                  ╰───────────── remove this from nativeBuildInputs
>     │                                  │             
>     │                                  ╰───────────── part of this list
> ────╯
> ```

Oh right, I forgot about this.

**Review by natsukium — APPROVED** [review]
LGTM

**Comment by GaetanLepage** [comment]
> LGTM

Thank you for the review !

**Pull request merged**